### PR TITLE
Upgrade numpy and fix test

### PR DESCRIPTION
**Pull request by isaranto**
Upgrading numpy version to match kserve 0.9.0 [requirements](https://github.com/kserve/kserve/blob/master/python/kserve/requirements.txt#L11)

- Fix failing test
- Update changelog
- Remove upper bound restriction from scipy requirement

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 # Changelog
 All notable changes to this project will be documented in this file.
 
+## [2.11.10]
```
Comment (elukey) [inline CHANGELOG.md:4]: Should we also add a note about the scipy's bump?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
+Bump numpy version to 1.24 and fix failing test (numpy.float has been deprecated)
+
+## [2.11.9]
+* Add github actions for CI with debian bullseye image
+* updated dictionaries that have a hunspell package in debian. hunspell is a "superset" of myspell.
```
Comment (elukey) [inline CHANGELOG.md:9]: Nit: uppercase

**Review by elukey — COMMENTED** [review]
Left two nits, but all LGTM!

File: CHANGELOG.md (hunk 2 above)
Comment (isaranto) [reply CHANGELOG.md]: 579341605b186561b4b53f1185e61c7ccaa984c9 Done!

File: CHANGELOG.md (hunk 1 above)
Comment (isaranto) [reply CHANGELOG.md:4]: 579341605b186561b4b53f1185e61c7ccaa984c9 Done!

**Review by elukey — APPROVED** [review]

**Pull request merged**